### PR TITLE
BNH-127 - add AdminControllerTests

### PR DESCRIPTION
**Pull request by cec200wire**
I've added unit tests for all the endpoints in the Admin Controller. I've also done some minor refactoring in the Admin Controller, and to a lesser extent the Admin Service - this was primarily just changing the order in which the endpoints/methods appear, in order to group together ones which are to do with similar issues. I've been finding it increasingly difficult to work out if a relevant method already exists or not, since everything was a bit jumbled up - hopefully having re-ordered things slightly will make this easier for everyone.

### CODE REVIEW

File: web/Services/AdminService.cs [hunk 1]
```diff
@@ -44,11 +57,43 @@ public void CancelAdminAccessRequest(BricksAndHeartsUser user)
...
 
     public async Task<(List<UserDbModel> CurrentAdmins, List<UserDbModel> PendingAdmins)> GetAdminLists()
     {
         return (await GetCurrentAdmins(), await GetPendingAdmins());
     }
+    
+    private async Task<List<UserDbModel>> GetCurrentAdmins()
+    {
+        var currentAdmins = await _dbContext.Users.Where(u => u.IsAdmin == true).ToListAsync();
+        return currentAdmins;
```
Comment (BenRamchandani) [inline web/Services/AdminService.cs:88]: Not your code, but anyway
```suggestion
        return await _dbContext.Users.Where(u => u.IsAdmin == true).ToListAsync();
```

File: web/Services/AdminService.cs [hunk 2]
```diff
@@ -44,11 +57,43 @@ public void CancelAdminAccessRequest(BricksAndHeartsUser user)
...
     }
+    
+    public void ApproveAdminAccessRequest(int userId)
+    {
+        var userToAdmin = _dbContext.Users.Single(u => u.Id == userId);
+        
+        userToAdmin.IsAdmin = true;
+        userToAdmin.HasRequestedAdmin = false;
+        _dbContext.SaveChanges();
+        _logger.LogInformation("Admin request approved");
```
Comment (BenRamchandani) [inline web/Services/AdminService.cs:68]: Could you log the user id that we're making an admin as part of this message? or even better move the log to the controller and log both the user we're changing and the current user

File: web/Controllers/AdminController.cs [hunk 3]
```diff
@@ -68,15 +67,49 @@ public IActionResult CancelAdminAccessRequest()
...
+                "Already an admin"));
+    }
+    
+    [Authorize(Roles = "Admin")]
+    [HttpPost]
+    public ActionResult AcceptAdminRequest(int userToAcceptId)
+    {
+        if (_adminService.GetUserFromId(userToAcceptId) == null)
+        {
+            return View("Error", new ErrorViewModel());
```
Comment (BenRamchandani) [inline web/Controllers/AdminController.cs:85]: Not your code again, but tbh I think this should just throw an exception - that will at least log something. I think omit this check completely and let the service method throw an exception (and same for reject)

**Review by BenRamchandani — DISMISSED** [review]
Tests look good, though I've thought of more refactors/changes if you don't mind...

File: web/Services/AdminService.cs [hunk 4]
```diff
@@ -44,11 +55,47 @@ public void CancelAdminAccessRequest(BricksAndHeartsUser user)
         userRecord.HasRequestedAdmin = false;
         _dbContext.SaveChanges();
     }
+    
+    public void ApproveAdminAccessRequest(int userId)
+    {
+        var userToAdmin = _dbContext.Users.SingleOrDefault(u => u.Id == userId);
+        if (userToAdmin == null)
+        {
+            throw new Exception();
```
Comment (BenRamchandani) [inline web/Services/AdminService.cs:64]: Worth having some text in there for someone debugging it (same below)
```suggestion
            throw new Exception($"No user found with id {userId}");
```

**Review by BenRamchandani — APPROVED** [review]

**Comment by cec200wire** [comment]
I had to rebase again, due to merge conflicts with my own work on another branch.

**Comment by cec200wire** [comment]
Git did something weird and tangled up two of my current branches - this hasn't yet rebased properly. I'll try again.

**Review by BenRamchandani — APPROVED** [review]
Still slightly mixed up, the table clickable changes are in here too, but fine by me to just merge anyway

**Pull request merged**